compiler/rustc_data_structures/src/vec_cache/tests.rs

```diff
@@ -0,0 +1,96 @@
+use super::*;
+
+#[test]
+#[cfg(not(miri))]
+fn vec_cache_empty() {
+    let cache: VecCache<u32, u32, u32> = VecCache::default();
+    for key in 0..u32::MAX {
+        assert!(cache.lookup(&key).is_none());
+    }
+}
+
+#[test]
+fn vec_cache_insert_and_check() {
+    let cache: VecCache<u32, u32, u32> = VecCache::default();
+    cache.complete(0, 1, 2);
+    assert_eq!(cache.lookup(&0), Some((1, 2)));
+}
+
+#[test]
+fn sparse_inserts() {
+    let cache: VecCache<u32, (), u32> = VecCache::default();
+    // each slot is 4 bytes, so 2^25 entries (on non-virtual memory systems, like e.g. Windows) will
+    // mean 128 megabytes of allocated memory. Going beyond that is probably not reasonable for
+    // tests.
+    for power in 0..25 {
+        let key = 1u32 << power;
+        cache.complete(key, (), key);
```

Comment:
can you also add a `lookup` here to make sure lookup is also correct for later buckets?